libkmod/libkmod.c

```diff
@@ -444,6 +445,15 @@ static char *lookup_builtin_file(struct kmod_ctx *ctx, const char *name)
 	return line;
 }
 
+static bool lookup_builtin_file(struct kmod_ctx *ctx, const char *name)
+{
+	char *line = lookup_file(ctx, KMOD_INDEX_MODULES_BUILTIN, name);
+	bool found = line != NULL;
+
+	free(line);
+	return found;
```

Comment:
You can simplify this function by using `_cleanup_free_ char *line` and just `return line != NULL`.

Reply:
That was my initial plan although scan-build (aka clang analyse) does not properly handle `__attribute__((cleanup))` and flags it as memory leak.

I am aiming to get ^^ as part of the CI and this gets us a step closer.

Reply:
We have this construct already in code base, see: https://github.com/kmod-project/kmod/blob/master/libkmod/libkmod.c#L498

I don't like the idea to keep existing features in a more expressive way just to make an analyzer happy, since it must complain about the existing one already, too.

Reply:
Yes, I'm aware that we have ~dozen instances in-tree. Pretty much all of those get flagged by clang-analyse.

Similar to the analysers and generally having wider compiler coverage - there is more value in biting the bullet "fixing" a few instances and keeping the tools happy IMHO.

Ideally clang will get fixed - if you want to give it a go, that would be deeply appreciated.

Reply:
So do you plan to remove the existing feature? Right now, it's a mix. Doesn't seem worthwhile for anyone.

Reply:
nack on removal